ray-operator/controllers/ray/common/pod.go

```diff
@@ -160,6 +187,40 @@ func BuildPod(podTemplateSpec v1.PodTemplateSpec, rayNodeType rayiov1alpha1.RayN
 
 	setContainerEnvVars(&pod.Spec.Containers[rayContainerIndex], rayNodeType, rayStartParams, svcName)
 
+	// add health check for head node
+
+	// health check only works for 1.13
+	if major, minor, _, err := ParseVersion(rayVersion); err == nil && major*100+minor >= 113 {
+		if rayNodeType == rayiov1alpha1.HeadNode {
+			probe := &v1.Probe{}
+			if pod.Spec.Containers[rayContainerIndex].ReadinessProbe == nil {
+				pod.Spec.Containers[rayContainerIndex].ReadinessProbe = probe
+			}
+			probe = pod.Spec.Containers[rayContainerIndex].ReadinessProbe
+			if probe.Exec == nil && probe.HTTPGet == nil && probe.TCPSocket == nil {
+				probe.Exec = &v1.ExecAction{Command: []string{
+					"python", "-c",
+					"from ray._private.gcs_utils import check_health;import sys;" +
```

Comment:
Can we use the CLI command `ray health-check` (which in turn calls `ray._private.gcs_utils`?)

There's a nontrivial probability of burning ourselves in the future if we import the private function :)

Reply:
I think the idea here is to use `curl`?

Reply:
CLI ExecAction is fine -- we don't necessarily need an http endpoint.

Just replace the whole string with "ray health-check" -- it does the same thing.

Reply:
Ok, I found the relevant discussion in the doc.
https://docs.google.com/document/d/1xAeQotz2_n3scsTRI3-RGBAFGTUbyclIxEX78INV1cg/edit?disco=AAAAbvh3Jm8
However you decide to structure the health-check, let's not use a private function in the final version.

Reply:
This part is updated now. We use the ones suggested by @brucez-anyscale and @iycheng